include/boost/geometry/formulas/cartesian.hpp

```diff
@@ -0,0 +1,137 @@
+// Boost.Geometry
+
+// Copyright (c) 2019 Oracle and/or its affiliates.
+
+// Contributed and/or modified by Vissarion Fysikopoulos, on behalf of Oracle
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_FORMULAS_CARTESIAN_HPP
+#define BOOST_GEOMETRY_FORMULAS_CARTESIAN_HPP
+
+#include <boost/concept_check.hpp>
+#include <boost/core/ignore_unused.hpp>
+#include <boost/mpl/if.hpp>
+#include <boost/type_traits/is_void.hpp>
+
+#include <boost/geometry/core/access.hpp>
+#include <boost/geometry/core/point_type.hpp>
+
+#include <boost/geometry/algorithms/convert.hpp>
+#include <boost/geometry/arithmetic/arithmetic.hpp>
+#include <boost/geometry/arithmetic/dot_product.hpp>
+
+#include <boost/geometry/util/select_coordinate_type.hpp>
+
+// Helper geometry (projected point on line)
+#include <boost/geometry/geometries/point.hpp>
+
+namespace boost { namespace geometry { namespace formula
+{
+
+template <
+    typename CalculationType,
+    bool EnableClosestPoint = false
+>
+class comparable_cartesian_point_segment_distance
+{
+
+public :
+
+    struct result_type
+    {
+        result_type()
+            : distance(0)
+            , x1(0)
+            , y1(0)
+            , x2(0)
+            , y2(0)
+        {}
+
+        CalculationType distance;
+        CalculationType x1;
+        CalculationType y1;
+        CalculationType x2;
+        CalculationType y2;
+    };
+
+    template <typename Point, typename PointOfSegment, typename Strategy>
+    static inline result_type
+    apply(Point const& p,
+          PointOfSegment const& p1,
+          PointOfSegment const& p2,
+          Strategy const& comparable_distance_strategy)
```

Comment:
If this uses a strategy I'm not sure if `formulas` is the correct place for it. It looks more like this should be somewhere in `strategies`, since this is also the result type which some of the strategies can return, right? And in the worst case it should probably be a `detail` of an algorithm.

Reply:
I moved to strategies, since detail is not CS-specific.